crates/daphne/src/roles/helper/mod.rs

```diff
@@ -11,21 +11,34 @@ use crate::{
     constants::DapMediaType,
     error::DapAbort,
     messages::{
-        constant_time_eq, AggregateShare, AggregateShareReq, AggregationJobInitReq,
-        PartialBatchSelector, TaskId,
+        constant_time_eq,
+        request::{AggregationJobRequestHash, HashedAggregationJobReq},
+        AggregateShare, AggregateShareReq, AggregationJobId, PartialBatchSelector, TaskId,
     },
     metrics::{DaphneRequestType, ReportStatus},
     protocol::aggregator::ReplayProtection,
-    DapAggregationParam, DapError, DapRequest, DapResponse, DapTaskConfig,
+    DapAggregationParam, DapError, DapRequest, DapResponse, DapTaskConfig, DapVersion,
 };
 
 /// DAP Helper functionality.
 #[async_trait]
-pub trait DapHelper: DapAggregator {}
+pub trait DapHelper: DapAggregator {
+    /// Asserts that either:
+    /// - this the first time we see this aggregation job
+    /// - this aggregation job has been seen before and it hasn't changed since the last time we
+    ///     saw it.
+    async fn assert_agg_job_is_immutable(
```

Comment:
This is better aligned with the terminology used elsewhere.

```suggestion
    async fn assert_agg_job_req_is_legal(
```